src/mongo/mongoConnectionStrings.ts

```diff
@@ -64,8 +64,27 @@ export async function getHostPortFromConnectionString(connectionString: string):
         port = serverConfig['port'];
     }
 
-    return {
-        host: host,
-        port: port
-    };
+    return new ParsedMongoConnectionString(connectionString, host, port, getDatabaseNameFromConnectionString(connectionString));
+}
+
+export class ParsedMongoConnectionString {
+    public readonly host: string;
+    public readonly port: string;
+    public readonly databaseName: string;
```

Comment:
databaseName: string|undefined

Also please add comment that no db means account level

Reply:
Aren't some of the others optional, too?

Reply:
Urgg stictNullChecks would've caught this for me :P 

And I think the rest are required.